tests/unit_tests/steps/test_analyse.py

```diff
@@ -44,15 +44,8 @@ def test_duplicate_symbol(self,
         """
         analysed_files[1].symbol_defs.add('foo_1')
 
-        with warns(UserWarning):
-            result = _gen_symbol_table(analysed_files=analysed_files)
-
-        assert result == {
-            'foo_1': Path('foo.c'),
-            'foo_2': Path('foo.c'),
-            'bar_1': Path('bar.c'),
-            'bar_2': Path('bar.c'),
-        }
+        with raises(RuntimeError):
```

Comment:
It is common all over the Fab code but `RuntimeError` is something we want to get away from. It's too generic. In this case `ValueError` may be more appropriate but we have `FabException` (`from fab import FabException`) available.